File: Converter/app/src/main/java/com/example/converter/ResponceModel.java
package com.example.converter;

import java.util.ArrayList;

public class ResponceModel {
    public class Rates {
        private double CAD;

        private double HKD;

        private double ISK;

        private double PHP;

        private double DKK;

        private double HUF;

        private double CZK;

        private double AUD;

        private double RON;

        private double SEK;

        private double IDR;

        private double INR;

        private double BRL;

        private double RUB;

        private double HRK;

        private double JPY;

        private double THB;

        private double CHF;

        private double SGD;

        private double PLN;

        private double BGN;

        private double TRY;

        private double CNY;

        private double NOK;

        private double NZD;

        private double ZAR;

        private double USD;

        private double MXN;

        private double ILS;

        private double GBP;

        private double KRW;

        private double MYR;

        public ConverterRow getCAD() {
            return new ConverterRow("CAD", this.CAD);
        }

        public ConverterRow getHKD() {
            return new ConverterRow("HKD", this.HKD);
        }

        public ConverterRow getISK() {
            return new ConverterRow("ISK", this.ISK);
        }

        public ConverterRow getPHP() {
            return new ConverterRow("PHP", this.PHP);
        }

        public ConverterRow getDKK() {
            return new ConverterRow("DKK", this.DKK);
        }

        public ConverterRow getHUF() {
            return new ConverterRow("HUF", this.HUF);
        }

        public ConverterRow getCZK() {
            return new ConverterRow("CZK", this.CZK);
        }

        public ConverterRow getAUD() {
            return new ConverterRow("AUD", this.AUD);
        }

        public ConverterRow getRON() {
            return new ConverterRow("RON", this.RON);
        }

        public ConverterRow getSEK() {
            return new ConverterRow("SEK", this.SEK);
        }

        public ConverterRow getIDR() {
            return new ConverterRow("IDR", this.IDR);
        }

        public ConverterRow getINR() {
            return new ConverterRow("INR", this.INR);
        }

        public ConverterRow getBRL() {
            return new ConverterRow("BRL", this.BRL);
        }

        public ConverterRow getRUB() {
            return new ConverterRow("RUB", this.RUB);
        }

        public ConverterRow getHRK() {
            return new ConverterRow("HRK", this.HRK);
        }

        public ConverterRow getJPY() {
            return new ConverterRow("JPY", this.JPY);
        }

        public ConverterRow getTHB() {
            return new ConverterRow("THB", this.THB);
        }

        public ConverterRow getCHF() {
            return new ConverterRow("CHF", this.CHF);
        }

        public ConverterRow getSGD() {
            return new ConverterRow("SGD", this.SGD);
        }

        public ConverterRow getPLN() {
            return new ConverterRow("PLN", this.PLN);
        }

        public ConverterRow getBGN() {
            return new ConverterRow("BGN", this.BGN);
        }

        public ConverterRow getTRY() {
            return new ConverterRow("TRY", this.TRY);
        }

        public ConverterRow getCNY() {
            return new ConverterRow("CNY", this.CNY);
        }

        public ConverterRow getNOK() {
            return new ConverterRow("NOK", this.NOK);
        }

        public ConverterRow getNZD() {
            return new ConverterRow("NZD", this.NZD);
        }

        public ConverterRow getZAR() {
            return new ConverterRow("ZAR", this.ZAR);
        }

        public ConverterRow getUSD() {
            return new ConverterRow("USD", this.USD);
        }

        public ConverterRow getMXN() {
            return new ConverterRow("MXN", this.MXN);
        }

        public ConverterRow getILS() {
            return new ConverterRow("ILS", this.ILS);
        }

        public ConverterRow getGBP() {
            return new ConverterRow("GBP", this.GBP);
        }

        public ConverterRow getKRW() {
            return new ConverterRow("KRW", this.KRW);
        }

        public ConverterRow getMYR() {
            return new ConverterRow("MYR", this.MYR);
        }
    }

    private Rates rates;

    private String base;

    private String date;

    public ConverterRow getEUR() {
        return new ConverterRow(this.base, (double) 1);
    }

    public String getDate() {
        return this.date;
    }

    public ArrayList<ConverterRow> GetBase() { //ArrayList на случай изменения количества валют в Json ответе, чтобы можно было за несколько строк все обновить
        ArrayList<ConverterRow> Base = new ArrayList<ConverterRow>();//можно было бы не делать этот класс, а обрабатывать Json ответ как строку, но я посчитал, что мне будет так удобнее
        Base.add(getEUR());
        Base.add(rates.getCAD());
        Base.add(rates.getHKD());
        Base.add(rates.getISK());
        Base.add(rates.getPHP());
        Base.add(rates.getDKK());
        Base.add(rates.getHUF());
        Base.add(rates.getCZK());
        Base.add(rates.getAUD());
        Base.add(rates.getRON());
        Base.add(rates.getSEK());
        Base.add(rates.getIDR());
        Base.add(rates.getINR());
        Base.add(rates.getBRL());
        Base.add(rates.getRUB());
        Base.add(rates.getHRK());
        Base.add(rates.getJPY());
        Base.add(rates.getTHB());
        Base.add(rates.getCHF());
        Base.add(rates.getSGD());
        Base.add(rates.getPLN());
        Base.add(rates.getBGN());
        Base.add(rates.getTRY());
        Base.add(rates.getCNY());
        Base.add(rates.getNOK());
        Base.add(rates.getNZD());
        Base.add(rates.getZAR());
        Base.add(rates.getUSD());
        Base.add(rates.getMXN());
        Base.add(rates.getILS());
        Base.add(rates.getGBP());
        Base.add(rates.getKRW());
        Base.add(rates.getMYR());
        return Base;
    }
}

File: Converter3/app/src/main/res/ResponceModel.kt
package com.example.converter2

import java.util.*

class ResponceModel {
    inner class Rates {
        var CAD = 0.0
        var HKD = 0.0
        var ISK = 0.0
        var PHP = 0.0
        var DKK = 0.0
        var HUF = 0.0
        var CZK = 0.0
        var AUD = 0.0
        var RON = 0.0
        var SEK = 0.0
        var IDR = 0.0
        var INR = 0.0
        var BRL = 0.0
        var RUB = 0.0
        var HRK = 0.0
        var JPY = 0.0
        var THB = 0.0
        var CHF = 0.0
        var SGD = 0.0
        var PLN = 0.0
        var BGN = 0.0
        var TRY = 0.0
        var CNY = 0.0
        var NOK = 0.0
        var NZD = 0.0
        var ZAR = 0.0
        var USD = 0.0
        var MXN = 0.0
        var ILS = 0.0
        var GBP = 0.0
        var KRW = 0.0
        var MYR = 0.0
        fun getCAD(): ConverterRow {
            return ConverterRow("CAD", CAD);
        }

        fun getHKD(): ConverterRow {
            return ConverterRow("HKD", HKD);
        }

        fun getISK(): ConverterRow {
            return ConverterRow("ISK", ISK);
        }

        fun getPHP(): ConverterRow {
            return ConverterRow("PHP", PHP);
        }

        fun getDKK(): ConverterRow {
            return ConverterRow("DKK", DKK);
        }

        fun getHUF(): ConverterRow {
            return ConverterRow("HUF", HUF);
        }

        fun getCZK(): ConverterRow {
            return ConverterRow("CZK", CZK);
        }

        fun getAUD(): ConverterRow {
            return ConverterRow("AUD", AUD);
        }

        fun getRON(): ConverterRow {
            return ConverterRow("RON", RON);
        }

        fun getSEK(): ConverterRow {
            return ConverterRow("SEK", SEK);
        }

        fun getIDR(): ConverterRow {
            return ConverterRow("IDR", IDR);
        }

        fun getINR(): ConverterRow {
            return ConverterRow("INR", INR);
        }

        fun getBRL(): ConverterRow {
            return ConverterRow("BRL", BRL);
        }

        fun getRUB(): ConverterRow {
            return ConverterRow("RUB", RUB);
        }

        fun getHRK(): ConverterRow {
            return ConverterRow("HRK", HRK);
        }

        fun getJPY(): ConverterRow {
            return ConverterRow("JPY", JPY);
        }

        fun getTHB(): ConverterRow {
            return ConverterRow("THB", THB);
        }

        fun getCHF(): ConverterRow {
            return ConverterRow("CHF", CHF);
        }

        fun getSGD(): ConverterRow {
            return ConverterRow("SGD", SGD);
        }

        fun getPLN(): ConverterRow {
            return ConverterRow("PLN", PLN);
        }

        fun getBGN(): ConverterRow {
            return ConverterRow("BGN", BGN);
        }

        fun getTRY(): ConverterRow {
            return ConverterRow("TRY", TRY);
        }

        fun getCNY(): ConverterRow {
            return ConverterRow("CNY", CNY);
        }

        fun getNOK(): ConverterRow {
            return ConverterRow("NOK", NOK);
        }

        fun getNZD(): ConverterRow {
            return ConverterRow("NZD", NZD);
        }

        fun getZAR(): ConverterRow {
            return ConverterRow("ZAR", ZAR);
        }

        fun getUSD(): ConverterRow {
            return ConverterRow("USD", USD);
        }

        fun getMXN(): ConverterRow {
            return ConverterRow("MXN", MXN);
        }

        fun getILS(): ConverterRow {
            return ConverterRow("ILS", ILS);
        }

        fun getGBP(): ConverterRow {
            return ConverterRow("GBP", GBP);
        }

        fun getKRW(): ConverterRow {
            return ConverterRow("KRW", KRW);
        }

        fun getMYR(): ConverterRow {
            return ConverterRow("MYR", MYR);
        }

    }

    private var rates: Rates? = null
    private var base: String? = null
    var date: String? = null
    get() = date;
    fun getEUR():ConverterRow{ return ConverterRow(base!!, 1.toDouble())}

    fun GetBase(): ArrayList<ConverterRow> { //ArrayList на случай изменения количества валют в Json ответе, чтобы можно было за несколько строк все обновить
        val Base = ArrayList<ConverterRow>() //можно было бы не делать этот класс, а обрабатывать Json ответ как строку, но я посчитал, что мне будет так удобнее
        Base.add(getEUR())
        Base.add(rates!!.getCAD())
        Base.add(rates!!.getHKD())
        Base.add(rates!!.getISK())
        Base.add(rates!!.getPHP())
        Base.add(rates!!.getDKK())
        Base.add(rates!!.getHUF())
        Base.add(rates!!.getCZK())
        Base.add(rates!!.getAUD())
        Base.add(rates!!.getRON())
        Base.add(rates!!.getSEK())
        Base.add(rates!!.getIDR())
        Base.add(rates!!.getINR())
        Base.add(rates!!.getBRL())
        Base.add(rates!!.getRUB())
        Base.add(rates!!.getHRK())
        Base.add(rates!!.getJPY())
        Base.add(rates!!.getTHB())
        Base.add(rates!!.getCHF())
        Base.add(rates!!.getSGD())
        Base.add(rates!!.getPLN())
        Base.add(rates!!.getBGN())
        Base.add(rates!!.getTRY())
        Base.add(rates!!.getCNY())
        Base.add(rates!!.getNOK())
        Base.add(rates!!.getNZD())
        Base.add(rates!!.getZAR())
        Base.add(rates!!.getUSD())
        Base.add(rates!!.getMXN())
        Base.add(rates!!.getILS())
        Base.add(rates!!.getGBP())
        Base.add(rates!!.getKRW())
        Base.add(rates!!.getMYR())
        return Base
    }
}
